workflows/cloudify_system_workflows/snapshots/postgres.py

```diff
@@ -313,6 +314,73 @@ def restore_config_tables(self, config_path):
         ])
         self._restore_dump(new_dump_file, self._db_name)
 
+    def dump_status_reporter_users(self, tempdir):
+        ctx.logger.debug('Dumping status reporter users...')
+        path = os.path.join(tempdir, 'status_reporter_users.dump')
+        command = self.get_psql_command(self._db_name)
+
+        query = (
+            'select array_to_json(array_agg(row)) from ('
+            'select * from users where id in ({0})'
+            ') row;'.format(_STATUS_REPORTERS_IDS_QUERY_TUPLE)
+        )
+        command.extend([
+            '-c', query,
+            '-t',  # Dump just the data, without extra headers, etc
+            '-o', path,
+        ])
+        run_shell(command)
+        ctx.logger.debug('Dumped status reporter users to {}'.format(path))
+        return path
+
+    def dump_status_reporter_roles(self, tempdir):
+        ctx.logger.debug('Dumping status reporter users\' roles')
+        path = os.path.join(tempdir, 'status_reporter_roles.dump')
+        command = self.get_psql_command(self._db_name)
+
+        query = (
+            'select array_to_json(array_agg(row)) from ('
+            'select * from users_roles where user_id in ({0})'
+            ') row;'.format(_STATUS_REPORTERS_IDS_QUERY_TUPLE)
+        )
+        command.extend([
+            '-c', query,
+            '-t',  # Dump just the data, without extra headers, etc
+            '-o', path,
+        ])
+        run_shell(command)
+        ctx.logger.debug('Dumped status reporter roles to {}'.format(path))
+        return path
+
+    @staticmethod
+    def _restore_json_dump_file(dump_path):
+        with open(dump_path) as dump_file:
+            return json.load(dump_file)
+
+    def restore_status_reporter_users(self, users_dump_path):
```

Comment:
I think restore_status_reporter_users and restore_status_reporter_roles functions are redundant and instead we can just call _restore_json_dump_file with the appropriate path